completer/README.md

```diff
@@ -0,0 +1,132 @@
+# Custom Completer
+
+> Provide a connector to customize tab completion results in a notebook.
+
+- [Code structure](#code-structure)
+- [Creating a custom connector](#creating-a-custom-connector)
+- [Aggregating connector responses](#aggregating-connector-responses)
+- [Disabling a JupyterLab plugin](#disabling-a-jupyterlab-plugin)
+- [Asynchronous extension initialization](#asynchronous-extension-initialization)
+- [Where to go next](#where-to-go-next)
+
+![Custom completion](preview.png)
+
+In this example, you will learn how to customize the behavior of JupyterLab notebooks' tab completion.
+
+## Code structure
+
+The code is split into three parts:
+
+1.  the JupyterLab plugin that activates all the extension components and connects
+    them to the main _JupyterLab_ application via commands,
+2.  a custom `CompletionConnector`, adapted from jupyterlab/packages/completer/src/connector.ts,
+    that aggregates completion results from three sources: _JupyterLab_'s existing `KernelConnector` and `ContextConnector`, plus...
+3.  `CustomConnector`, a lightweight source of mocked completion results.
+
+The first part is contained in the `index.ts` file, the second is in `connector.ts`, and the third is in `customconnector.ts`.
+
+## Creating a custom DataConnector
+
+In `src/customconnector.ts` we define a `CustomConnector` to generate mock autocomplete suggestions. Like the `ContextConnector` it is based on, `CustomConnector` extends _JupyterLab_'s [`DataConnector`](https://jupyterlab.readthedocs.io/en/latest/api/classes/statedb.dataconnector.html) class.
+
+- Try adding a new token to `tokenList` in the `completionHint` function.
+- Try modifying the `completionHint` function to return no matches when `token.value.length` is less than 3.
```

Comment:
Some sentence before this, like `This custom connector will behave like this:`

Could you change this part to describe the actual code doing those actions?